resources/util.ts

```diff
@@ -77,6 +79,66 @@ const jobToRoleMap: Map<Job, Role> = (() => {
   return map;
 })();
 
+type WatchCombatantFunc = (params: {
+  ids?: number[];
+  names?: string[];
+  props?: string[];
+  delay?: number;
+}, func: (ret: GetCombatantsRet) => boolean) => Promise<boolean>;
+
+const watchCombatantMap: { cancel: boolean }[] = [];
+
+const watchCombatant: WatchCombatantFunc = (params, func) => {
+  let resolver: (value: boolean) => void;
+  let rejecter: () => void;
```

Comment:
Why not just wrap all things in promise initialization?

Reply:
Originally I was storing off the rejecter in the watchCombatantMap, but that ended up not really working out. Now that it's not needed outside the scope of the promise, I've changed it as suggested.